core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -167,6 +168,11 @@ public MediaType getMediaType() {
     }
 
     public void updateFromOther(FeedMedia other) {
+        // we try to cover two cases: (1) feed did include file before (2) feed contained wrong URL
+        // if item.getAutoDownload() is false, the file has been downloaded before
+        if((TextUtils.isEmpty(download_url) || item.getAutoDownload()) && !TextUtils.isEmpty(other.download_url)) {
+            item.setNew();
```

Comment:
this might be silly, but we could always set it new if the download URL now exists in other (and is empty in this)?  AutoDownload == 0 doesn't necessarily mean it was downloaded before, could have been failing because the feed was bad.

Any reason we shouldn't set it to new if AutoDownload == 0?